### ISSUE TITLE
json.decoder.JSONDecodeError in venv_inspect

### ISSUE THREAD

**Issue by jaraco** [pypa/pipx#723]
**Describe the bug**

On a system that is running pipx on Python 3.9, Running pipx install on Python 3.10 fails with a JSONDecodeError.

```
bin $ python3.10 -m pipx install --verbose devpi-client
pipx >(setup:717): pipx version is 0.16.3
pipx >(setup:718): Default python interpreter is '/Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10'
pipx >(package_name_from_spec:323): Determined package name: devpi-client
pipx >(package_name_from_spec:324): Package name determined in 0.0s
pipx >(run_subprocess:135): running /Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10 -m venv --without-pip /Users/jaraco/.local/pipx/venvs/devpi-client
creating virtual environment...
pipx >(run_subprocess:135): running /Users/jaraco/.local/pipx/venvs/devpi-client/bin/python -c import sysconfig; print(sysconfig.get_path('purelib'))
pipx >(run_subprocess:135): running /Users/jaraco/.local/pipx/shared/bin/python -c import sysconfig; print(sysconfig.get_path('purelib'))
pipx >(run_subprocess:135): running /Users/jaraco/.local/pipx/venvs/devpi-client/bin/python --version
installing devpi-client...
pipx >(_parsed_package_to_package_or_url:128): cleaned package spec: devpi-client
pipx >(run_subprocess:135): running /Users/jaraco/.local/pipx/venvs/devpi-client/bin/python -m pip install devpi-client
pipx >(run_subprocess:135): running <fetch_info_in_venv commands>
Traceback (most recent call last):
  File "<string>", line 10, in <module>
TypeError: can only concatenate str (not "int") to str

pipx >(rmdir:43): removing directory /Users/jaraco/.local/pipx/venvs/devpi-client
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/__main__.py", line 14, in <module>
    sys.exit(cli())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/main.py", line 765, in cli
    return run_pipx_command(parsed_pipx_args)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/main.py", line 193, in run_pipx_command
    return commands.install(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/commands/install.py", line 60, in install
    venv.install_package(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv.py", line 244, in install_package
    self._update_package_metadata(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv.py", line 318, in _update_package_metadata
    venv_package_metadata = self.get_venv_metadata_for_package(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv.py", line 300, in get_venv_metadata_for_package
    venv_metadata = inspect_venv(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv_inspect.py", line 232, in inspect_venv
    (venv_sys_path, venv_env, venv_python_version) = fetch_info_in_venv(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv_inspect.py", line 206, in fetch_info_in_venv
    venv_info = json.loads(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Switching back to Python 3.9, I'm able to install just fine.

**Comment by jaraco** [comment]
Okay, I found the issue. It's not because it's Python 3.10, but because it's not a final version.

```
bin $ python3.10 -m pdb -m pipx install devpi-client
> /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/__main__.py(1)<module>()
-> import os
(Pdb) b pipx/venv_inspect.py:206
Breakpoint 1 at /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv_inspect.py:206
(Pdb) c
> /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pipx/venv_inspect.py(206)fetch_info_in_venv()
-> venv_info = json.loads(
(Pdb) run_subprocess([venv_python_path, "-c", command_str], capture_stderr=False, log_cmd_str="<fetch_info_in_venv commands>",).stdout
Traceback (most recent call last):
  File "<string>", line 10, in <module>
TypeError: can only concatenate str (not "int") to str
''
(Pdb) print(command_str)

import json
import os
import platform
import sys

impl_ver = sys.implementation.version
implementation_version = "{0.major}.{0.minor}.{0.micro}".format(impl_ver)
if impl_ver.releaselevel != "final":
    implementation_version += impl_ver.releaselevel[0] + impl_ver.serial

sys_path = sys.path
try:
    sys_path.remove("")
except ValueError:
    pass

print(
    json.dumps(
        {
            "sys_path": sys_path,
            "python_version": "{0.major}.{0.minor}.{0.micro}".format(sys.version_info),
            "environment": {
                "implementation_name": sys.implementation.name,
                "implementation_version": implementation_version,
                "os_name": os.name,
                "platform_machine": platform.machine(),
                "platform_release": platform.release(),
                "platform_system": platform.system(),
                "platform_version": platform.version(),
                "python_full_version": platform.python_version(),
                "platform_python_implementation": platform.python_implementation(),
                "python_version": ".".join(platform.python_version_tuple()[:2]),
                "sys_platform": sys.platform,
            },
        }
    )
)

(Pdb) import sys
(Pdb) sys.implementation.version
sys.version_info(major=3, minor=10, micro=0, releaselevel='candidate', serial=1)
(Pdb) impl_ver = sys.implementation.version
(Pdb) impl_ver.releaselevel[0] + impl_ver.serial
*** TypeError: can only concatenate str (not "int") to str
```

**Comment by jaraco** [comment]
So apparently this was fixed in #706, but because no release was cut, I got to experience the troubleshooting steps too :(. I'd even searched for the error message, which wasn't present in the PR :(.

Please cut a release. Thanks.

**Comment by gaborbernat** [comment]
https://github.com/pypa/pipx/actions/runs/1100481144
